src/Illuminate/View/Compilers/ComponentTagCompiler.php

```diff
@@ -687,6 +705,60 @@ protected function parseComponentTagClassStatements(string $attributeString)
         );
     }
 
+    /**
+     * Parse @attributes statements in a given attribute string into their fully-qualified syntax.
+     *
+     * @param  string  $attributeString
+     * @return string
+     */
+    protected function parseComponentTagAttributesStatements(string $attributeString)
+    {
+        return preg_replace_callback(
+            '/@(attributes)(\( ( (?>[^()]+) | (?2) )* \))/x', function ($match) {
+
+
+                if ($match[1] === 'attributes') {
+                    return \Illuminate\Support\Arr::toHtmlAttributes(
+                        $this->reformatAttributeExpressionStringToArray($match[2])
+                    );
+                }
+
+                return $match[0];
+            }, $attributeString
+        );
+    }
+
+    /**
+     * Take an attribute string in expression format (surrounded by ([]))
+     * and reformat it into a compiled HTML attribute string.
+     *
+     * @param string $expression
+     * @return array
+     */
+    public function reformatAttributeExpressionStringToArray(string $expression)
```

Comment:
@timacdonald could use some pointers here if you don't mind?

For any `@attributes` call on a normal HTML element i'm just doing [this](https://github.com/laravel/framework/pull/52783/files#diff-19417a9fd60ee9c30ff3c669fd762e8a44e573b5392f0d46ece856c9bff1e4dfR16) - which is fairly simple.

For `@attributes` calls on slots and custom components, it's tough, because we can't compile _into_ an existing attribute and prefix with `:` in the same way we can with `@class`, the attributes need to be parsed before compile time, so that they're fully parsed and passed down into the component.

BUT, this doesn't work when functions and 

I'd ended up with this, which is taking the `@attributes` expression, parsing it using a regex, and turning it back into an array so it can be parsed properly by the new `toHtmlAttributes` method, but this isn't going to support things like calls to `when` or `$variables` properly.

Am i on the right track, or should i look at taking another approach? I'm trying to get my head around blade's compilation order, and the expectations of the output.

Cheers!

Reply:
Hey, Dan, I don't know a heap about the Blade compiler unfortunately.

I don't imagine that the regex is gonna be the answer...but then again blade is entirely regex so maybe it is?

Reply:
No worries man, i'll keep plucking at the problem, it feels like this is over-engineered to me, but will keep digging. 👍

Reply:
Sounds good. Don't take my feedback as anything you must do. Just sharing an opinion.